src/sqlfluff/testing/rules.py

```diff
@@ -142,3 +142,7 @@ def rules__test_helper(test_case):
         # If a `fixed` value is provided then check it matches
         if test_case.fix_str:
             assert res == test_case.fix_str
+        else:
+            # Check that tests without a fix_str do not apply any fixes.
+            # If the fix is expected, add the missing fix_str in the test case.
+            assert res == test_case.fail_str
```

Comment:
Indeed. How about this (info is in assert msg instead of comments):
```suggestion
            # check that tests without a fix_str do no apply ANY fixes
            assert res == test_case.fail_str, (
                "No fix_str was provided, but the rule modified the SQL. "
                "If the modification is expected, please add a fix_str to the test."
            )
```
I added "If the modification is expected" ^

Reply:
To be honest that reads a little off to me. I think it's better without it. Also noticed a slight typo in comment "do not" instead of "do no".

```suggestion
            # Check that tests without a fix_str do not apply any fixes.
            assert res == test_case.fail_str, (
                "No fix_str was provided, but the rule modified the SQL. "
                "Please add a fix_str to the test."
            )
```

Reply:
I'm just hesitating to instruct simply to add a fix_str. Do you think it's so obvious/implicit that anyone knows that fix_str should be added only if the fix result is as desired? Maybe it is.

Reply:
Altered by suggestion above. WDYT?